File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03828/p03828_num6_parsed.py
import math

N = int(input())

# calculate N!
fac = math.factorial(N)

# iterate through all numbers from 1 to N!
# check if each number is a positive divisor
count = 0
for i in range(1, fac+1):
    if fac % i == 0:
        count += 1

# print the number of positive divisors modulo 10^9+7
print(count % (10**9 + 7))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03828/p03828_num6_parsed.cpp
#include <iostream>
using namespace std;

const int MOD = 1000000007;

int main() {
    int N;
    cin >> N;
    
    // Compute the factorial of N
    long long fact = 1;
    for (int i = 1; i <= N; i++) {
        fact = (fact * i) % MOD;
    }
    
    // Compute the number of divisors
    int divisors = 1;
    for (int i = 2; i * i <= fact; i++) {
        int count = 0;
        while (fact % i == 0) {
            count++;
            fact /= i;
        }
        divisors = (divisors * (count + 1)) % MOD;
    }
    if (fact > 1) {
        divisors = (divisors * 2) % MOD;
    }
    
    cout << divisors << endl;
    
    return 0;
}